spyder/plugins/editor/widgets/codeeditor.py

```diff
@@ -2872,6 +2883,63 @@ def run_pygments_highlighter(self):
         if isinstance(self.highlighter, sh.PygmentsSH):
             self.highlighter.make_charlist()
 
+    def left_line(self, position, source_button):
+        """Process a line at position 'position' on leaving it.
+
+        Remove trailing whitespace on leaving a non-string line containing it.
+        This could also do other processing in the future based on the button
+        pressed ('source') or settings."""
+
+        # Get current position
+        cursor = self.textCursor()
+        current_pos = cursor.position()
+
+        # Chech if still on the line
+        cursor.setPosition(position)
+        line_range = (cursor.block().position(),
+                      cursor.block().position()
+                      + cursor.block().length() - 1)
+
+        def pos_in_line(pos):
+            if pos is None:
+                return False
+            return pos >= line_range[0] and pos <= line_range[1]
+
+        if pos_in_line(current_pos):
+            return
+
+        # Check if end of line in comment or string

```

Comment:
I don't remember why we arn't removing the trailing spaces in strings and comments but PEP8 still complains about them.

Reply:
It is removing the trailing spaces in comments, as it should (I don't see why they should be any different than anything but strings), so this should be corrected to indicate that correctly.

Strings are a different matter, as trailing whitespace can be meaningful in them if the string is not terminated on that line. Therefore, the safe course of action is not to remove it (which this feature currently does not, as it should) and have the user deal with it themselves (if they want all trailing whitespace removed no matter what, they can turn on "Remove trailing whitespace on save").

Reply:
> Strings are a different matter, as trailing whitespace can be meaningful in them if the string is not terminated on that line

It's a bad idea to have meaningful trailing whitespace even in a string literal but I can see that some people would want to do this. That code will however have a style error.

Reply:
@impact27 The trailing spaces are being striped from `"""string   ` in my testing

Reply:
> It's a bad idea to have meaningful trailing whitespace even in a string literal

I agree, but because it can have a functional effect, its better to not silently strip it and allow the user to see the PEP 8 warning and decide what to do.

> The trailing spaces are being striped from `"""string ` in my testing

Strange; I am not able to reproduce this behavior from that ``"""string`` or several other triple and single-quoted strings with ``Enter``, clicks or arrow keys, and as far as I can recall its worked pretty consistently throughout.

Reply:
Interesting. It seems to be consistently removed on my system 
![rstrip strings](https://user-images.githubusercontent.com/10513354/53068103-66710d80-3494-11e9-8358-1fa2caf8845a.gif)

Reply:
I am a bit confused. What I do is: 
1) set the position to the end of the line
2) call `self.in_comment_or_string`
3) If it is true, then stop
So it should not strip comments or strings
If it does then there probably is an error in `self.in_comment_or_string`

Reply:
> So it should not strip comments or strings

It most certainly does strip trailing whitespace from comments, as it well should—there's no rationale I'm aware of that would motivate not doing so. For strings, I can repro the behavior shown in @bcolsen 's screenshot—space is stripped in a string if on the same line as the opening ``"""`` (when it presumably shouldn't be), but is not stripped otherwise.